templates/deployments.yaml

```
@@ -43,6 +43,24 @@ spec:
       {{- end }}
       # Sidecar container for generating .htpasswd with st2 username & password pair and sharing produced file with the main st2auth container
       initContainers:
+      - name: wait-for-db
+        image: busybox:1.28
+        command:
+          - 'sh'
+          - '-c'
+          - >
+            until nc -z -w 2 {{ .Release.Name }}-mongodb-headless 27017 && echo mongodb ok;
+              do sleep 2;
+            done
```

```suggestion
            until nc -z -w 2 {{ .Release.Name }}-mongodb-headless 27017 && echo mongodb ok; do
              echo 'Waiting for MongoDB connection ...'
              sleep 2;
            done
```

Worth including a bit of logging there for users.

Same for RabbitMQ.

---

Done